components/lcd/esp_lcd_gc9503/include/esp_lcd_gc9503.h

```diff
@@ -10,26 +10,63 @@
 
 #pragma once
 
-#include "esp_lcd_types.h"
+#include <stdint.h>
+
+#include "hal/lcd_types.h"
+#include "esp_lcd_panel_vendor.h"
 #include "esp_lcd_panel_rgb.h"
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
 /**
- * @brief Create LCD panel for GC9503.
+ * @brief LCD panel initialization commands.
+ *
+ */
+typedef struct {
+    int cmd;                /*<! The specific LCD command */
+    const void *data;       /*<! Buffer that holds the command specific data */
+    size_t data_bytes;      /*<! Size of `data` in memory, in bytes */
+    unsigned int delay_ms;  /*<! Delay in milliseconds after this command */
+} gc9503_lcd_init_cmd_t;
```

Comment:
This is good idea, but I am thinking, if there is any solution, how to do it in `esp_lcd` or make any API in `esp_lcd`, which will help with this in all LCD drivers. I would like to see this structure in `esp_lcd`. 
Possible problems:
1. Some LCD drivers needs two byte commands
2. How to implement new API inside `esp_lcd` without breaking changes?

Reply:
I have discussed this idea with @suda-morris, and he didn't recommend putting this structure in `esp_lcd`.

1. The `data` has a `void *` type and can be used to pass any bytes commands. Special handling of different LCD commands can be performed internally within the driver.
2. There is no need to add a new API. The usage of this structure is like below, I'll add it into README:
```c
/**
 *  The array should be declared as static const and positioned outside the function.
 */
// static const gc9503_lcd_init_cmd_t lcd_init_cmds[] = {
// //   cmd   data        data_size  delay_ms
//    {0xf0, (uint8_t []){0xc3}, 1, 0},
//    {0xf0, (uint8_t []){0x96}, 1, 0},
//    {0xf0, (uint8_t []){0xa5}, 1, 0},
//     ...
// };

    esp_lcd_rgb_panel_config_t rgb_config= {
        .clk_src = LCD_CLK_SRC_PLL160M,
        .psram_trans_align = 64,
        .data_width = 16,
        .de_gpio_num = BSP_LCD_DE,
        .pclk_gpio_num = BSP_LCD_PCLK,
        .vsync_gpio_num = BSP_LCD_VSYNC,
        .hsync_gpio_num = BSP_LCD_HSYNC,
        .data_gpio_nums = {
            BSP_LCD_DATA0,
            BSP_LCD_DATA1,
            BSP_LCD_DATA2,
            BSP_LCD_DATA3,
            BSP_LCD_DATA4,
            BSP_LCD_DATA5,
            BSP_LCD_DATA6,
            BSP_LCD_DATA7,
            BSP_LCD_DATA8,
            BSP_LCD_DATA9,
            BSP_LCD_DATA10,
            BSP_LCD_DATA11,
            BSP_LCD_DATA12,
            BSP_LCD_DATA13,
            BSP_LCD_DATA14,
            BSP_LCD_DATA15,
        },
        .timings = GC9503_480_480_PANEL_60HZ_RGB_TIMING(),
        .flags.fb_in_psram = 1,
    };
    esp_lcd_panel_handle_t panel_handle = NULL;
    gc9503_vendor_config_t vendor_config = {
        .rgb_config = &rgb_config,
        .init_cmds = lcd_init_cmds,         // Only useful when `use_external_init_cmds` is set to 1
        .init_cmds_size = sizeof(lcd_init_cmds) / sizeof(gc9503_lcd_init_cmd_t),
        .flags = {
            .use_external_init_cmds = 1,    // Set to 1 if use specific initialization commands
        },
    };
    const esp_lcd_panel_dev_config_t panel_config = {
        ...
        .vendor_config = &vendor_config,
       ...
    };
    ESP_ERROR_CHECK(esp_lcd_new_panel_gc9503(io_handle, &panel_config, &panel_handle));
    ESP_ERROR_CHECK(esp_lcd_panel_reset(panel_handle));
    ESP_ERROR_CHECK(esp_lcd_panel_init(panel_handle));
```